op-challenger/challenger/output.go

```diff
@@ -0,0 +1,44 @@
+package challenger
+
+import (
+	"context"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+)
+
+var (
+	// supportedL2OutputVersion is the version of the L2 output that the challenger supports.
+	supportedL2OutputVersion = eth.Bytes32{}
+	// ErrInvalidBlockNumber is returned when the block number of the output does not match the expected block number.
+	ErrInvalidBlockNumber = errors.New("invalid block number")
+	// ErrUnsupportedL2OOVersion is returned when the output version is not supported.
+	ErrUnsupportedL2OOVersion = errors.New("unsupported l2oo version")
+)
+
+// ValidateOutput checks that a given output is expected via a trusted rollup node rpc.
+// It returns: if the output is correct, error
+func (c *Challenger) ValidateOutput(ctx context.Context, l2BlockNumber *big.Int, expected eth.Bytes32) (bool, *eth.Bytes32, error) {
```

Comment:
This function does three things:
1. Fetch data from the op-node
2. Validate the onchain data vs the op-node's view
3. Metrics & logging

I'd suggest that step 2 is a good candidate for pulling out into it's own function. That specific function is the primary business logic & can easily be tested in isolation & doesn't require lots of setup code to test.